lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/LeakDetector.java

```diff
@@ -155,19 +156,16 @@ private void checkField(UField field) {
                                             + "`" + referenced.getName() + "` pointing to `"
                                             + innerCls.getName() + "`); "
                                         + "this is a memory leak (and also breaks Instant Run)";
-                            report(field, modifierList, message);
+                            report(field, message);
                             break;
                         }
                     }
                 }
             }
         }
 
-        private void report(@NonNull PsiField field, @NonNull PsiModifierList modifierList,
-                @NonNull String message) {
-            Location location = mContext.getLocation(
-                    modifierList.getTextRange().getLength() > 0 ? modifierList : field);
-            mContext.report(ISSUE, field, location, message);
+        private void report(@NonNull UElement field, @NonNull String message) {
```

Comment:
Two ugly casts to UElement will be required to fix ambiguous method calls
`mContext.report(ISSUE, (UElement)field, mContext.getLocation((UElement) field), message);
`